Gems/AtomTressFX/Code/Rendering/HairCommon.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/RHIUtils.h>
+
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Image/StreamingImage.h>
+#include <Atom/RPI.Reflect/Buffer/BufferAssetView.h>
+
+// Hair specific
+#include <Rendering/HairCommon.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+
+            //!=====================================================================================
+            //!
+            //!                                Utility Functions
+            //!
+            //!=====================================================================================
+
+            // [To Do] examine if most of these functions can become global in RPI
+
+            //! Utility function to generate the Srg given the shader and the desired Srg name to be associated to.
+            //! If several shaders are sharing the same Srg (for example perView, perScene), it is enough to
+            //! create the Srg by associating it with a single shader and since the GPU signature and the data
+            //! are referring to the same shared description (preferable set in an [SrgDeclaration].aszli file)
+            //! The association with all shaders will work properly.
```

Comment:
Don't we want to avoid creating multiple instances of perScene SRG though? We should only be updating the one scene SRG per scene. Or am I missing something? Anyway, not really important for this PR, just wanted to ask the question.